Source/WebCore/rendering/RenderLayerCompositor.cpp

```diff
@@ -2608,6 +2608,17 @@ bool RenderLayerCompositor::attachWidgetContentLayers(RenderWidget& renderer)
     if (isCompositedPlugin(renderer)) {
         if (auto* contentsLayer = backing->layerForContents()) {
             addContentsLayerChildIfNecessary(*contentsLayer);
+
+            if (isLayerForPluginWithScrollCoordinatedContents(*layer)) {
+                if (auto pluginHostingNodeID = backing->scrollingNodeIDForRole(ScrollCoordinationRole::PluginHosting)) {
+                    auto* renderEmbeddedObject = dynamicDowncast<RenderEmbeddedObject>(renderer);
+                    if (auto pluginScrollingNodeID = renderEmbeddedObject->scrollingNodeID()) {
+                        if (auto* scrollingCoordinator = this->scrollingCoordinator())
+                            scrollingCoordinator->insertNode(ScrollingNodeType::PluginScrolling, pluginScrollingNodeID, pluginHostingNodeID, 0);
```

Comment:
This makes me think we might want a test that dumps the scrolling tree. We should also test dynamic changes to an `<embed>` src to and from PDF contents.

Reply:
Added this test, though cannot do "from" without just removing the <embed>